Jint/Engine.cs

```diff
@@ -42,6 +42,30 @@ public class Engine
 
         // cache of types used when resolving CLR type names
         internal Dictionary<string, Type> TypeCache = new Dictionary<string, Type>();
+		// cache of learned type conversions.
+		internal Dictionary<Type, Func<Engine, object, JsValue>> TypeMappers = new Dictionary<Type, Func<Engine, object, JsValue>>()
```

Comment:
What do you think about making it a static concurrent dictionary. That would make the list reusable across engine instances, which is ideal in a server scenario. This applies also to the other caches in this class.